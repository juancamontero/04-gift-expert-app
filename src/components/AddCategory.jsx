import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onChangeInput = (event) => {
    setInputValue(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    if (inputValue.trim().length <= 1) return; //para que no ejecute si no hay texto
    onNewCategory(inputValue.trim());
    setInputValue(""); // para borrar todo lo del input;
  };

  return (
    <form aria-label="add-cat-form" onSubmit={handleSubmit}>
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onChangeInput}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
