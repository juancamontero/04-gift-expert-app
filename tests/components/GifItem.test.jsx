import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe("<GifItem /> ", () => {
  const title = "Alfonso Ribeiro Dancing GIF";
  const url =
    "https://media3.giphy.com/media/IGpqj6DMmbeNy/giphy.gif?cid=5538b305w7xkjqdiaqah0n79sexyfrnvgt4qxsklh7crkjwy&rid=giphy.gif&ct=g";

  test("should snapShot match?", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("should the <img> has the correct URl and ALT ", () => {
    render(<GifItem title={title} url={url} />);

    // expect(screen.getByRole("img").src).toBe(url); //manera larga
    // expect(screen.getByRole("img").alt).toBe(title);//manera larga

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("should the TITLE be shown in the component", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
