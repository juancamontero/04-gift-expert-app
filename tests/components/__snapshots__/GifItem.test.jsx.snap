// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifItem />  should snapShot match? 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Alfonso Ribeiro Dancing GIF"
      src="https://media3.giphy.com/media/IGpqj6DMmbeNy/giphy.gif?cid=5538b305w7xkjqdiaqah0n79sexyfrnvgt4qxsklh7crkjwy&rid=giphy.gif&ct=g"
    />
    <p>
      Alfonso Ribeiro Dancing GIF
    </p>
  </div>
</div>
`;
